shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmap.java

```diff
@@ -899,4 +903,27 @@ void setBufferedImage(BufferedImage bufferedImage) {
   BufferedImage getBufferedImage() {
     return bufferedImage;
   }
+
+  void drawRect(RectF r, Paint paint) {
```

Comment:
Could you help to move this method under previous `drawRect`?

Reply:
yeah, i will do it..

Reply:
I'd recommend just having this method be:

```
  void drawRect(RectF r, Paint paint) {
    if (bufferedImage == null) {
      return;
    }
    Graphics2D graphics2D = bufferedImage.createGraphics();
    Rectangle2D r2d = new Rectangle2D.Float(r.left, r.top, r.right - r.left, r.bottom - r.top);
    graphics2D.draw(r2d);
    graphics2D.dispose();
  }
```

And not mucking with the underlying buffers.

Reply:
@hoisie the above code is not working as it is not using any color to draw it and it is failing tests..
So, we have to add either one statement into it as like : `graphics2D.setColor(paint.getColor());`
or we can fill pixels as i have done it, but it is not efficient..
I think we should add the above statement to set color for drawing..
I tested it and it's working..
There is one issue that graphics2D.setColor(Color color) accepts argument of Color type, whereas paint.getColor() returns int value.
I am looking to add it and then it willl work fine..

Reply:
@Akshay2131 what about trying `new Color(paint.getColor())`? The `Color` has a constructor with int color value as input: https://docs.oracle.com/javase/8/docs/api/java/awt/Color.html#Color-int-.

Reply:
I am thinking to do the same either using constructor or method..
I will add a constructor which takes int color value as input..

Reply:
Yeah, it's working properly..

Reply:
Whoops, this should work I believe:

  void drawRect(RectF r, Paint paint) {
    if (bufferedImage == null) {
      return;
    }
    Graphics2D graphics2D = bufferedImage.createGraphics();
    android.graphics.Color color = android.graphics.Color.valueOf(paint.getColor());
    graphics2D.setColor(new Color(color.red(), color.green(), color.blue(), color.alpha()));
    graphics2D.fill(new Rectangle2D.Float(r.left, r.top, r.right - r.left, r.bottom - r.top));
    graphics2D.dispose();
  }

Reply:
I added a line in the code you suggested earlier, and it's working now..
The code is as: 

`  void drawRect (RectF r, Paint paint) {

    if (bufferedImage == null) {
      return;
    }
    Graphics2D graphics2D = bufferedImage.createGraphics();
    Rectangle2D r2d = new Rectangle2D.Float(r.left, r.top, r.right - r.left, r.bottom - r.top);
    graphics2D.setColor(new Color(paint.getColor()));
    graphics2D.draw(r2d);
    graphics2D.dispose();
  } `

Reply:
I think the above code you suggested would also work..

Reply:
could you tell which additional test is to be added as pointed above by @utzcoz previously..

Reply:
> I vaguely recall reason that ShadowBitmap.drawRect(Rect r, Paint paint) does not use Graphics2d is because Graphics2d has slight behavior differences. I remember something like if you drawRect with something like 0x00ffffff, which is an 100% translucent white color, Graphics2d converts it to 0, whereas in real Android this remains '0x00ffffff', and this broke some tests. At least I think that's what it is. Maybe I should capture that in CTesque BitmapTest.

Is this corner case pointed out by @hoisie. @hoisie can accept we don't add test for at this PR and check it at later PR. It depends on you to add it or not. All occasions are good to me.

Reply:
Yeah it's fine not to handle or have a test for this corner case. We should try to defer as much to AWT as possible for this implementation.